go/cmd/dolt/commands/revert.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+
+	"github.com/dolthub/dolt/go/cmd/dolt/cli"
+	"github.com/dolthub/dolt/go/cmd/dolt/errhand"
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+	"github.com/dolthub/dolt/go/libraries/doltcore/env"
+	"github.com/dolthub/dolt/go/libraries/doltcore/merge"
+	"github.com/dolthub/dolt/go/libraries/utils/argparser"
+	"github.com/dolthub/dolt/go/libraries/utils/filesys"
+)
+
+var revertDocs = cli.CommandDocumentationContent{
+	ShortDesc: "Undo the changes introduced in a commit",
+	LongDesc: `Removes the changes made in a commit (or series of commits) from the working set, and then automatically commits the
+result. This is done by way of a three-way merge. If only one commit is specified (HEAD~1), then a merge is performed
+that is equivalent to the following:
+
+Base:   HEAD~1
+Ours:   Working Set
+Theirs: HEAD~2
```

Comment:
Definitely need that comment about a clean working set, good catch.

Although it is an implementation detail, `git revert` itself is a merge, and has the same user workflow as a merge. Error messages and other user-facing functionality also describes it as a merge, and I think that's to relate the workflow to a familiar one. I also think it helps the user to understand what to expect (for when we eventually do handle conflicts and constraint violations, it will be the exact same as with a merge). I'll update the doc, but I definitely want to keep the information about it being a merge.